python/tests/test_balance_metrics.py

```diff
@@ -32,6 +35,14 @@
 # we can remove this.
 
 
+def nodes_to_root(tree, u):
```

Comment:
How about this:
```python
path = []
u = tree.parent(u)
while u != tskit.NULL:
      paty.append(u)
      u = tree.parent(u)
return path
```

(This might be a useful library function in fact. Something like,
```python
def path(self, u, v=None):
     """
     Returns the path between two nodes u and v, such that ``len(tree.path(u, v)) == tree.path_length(u, v)``.
     If the second node ``v`` is not specified, return the path from u to root.
     """
```

Reply:
Let's just update the version of ``nodes_to_root`` here to use the implementation I suggested and put #2350 on the back-burner for now. Searching ``tree.roots`` on every loop iteration is unnecessarily inefficient when we can just check what the parent is in O(1) time.